config/README.md

```diff
@@ -5,9 +5,12 @@ this configuration, pass `--serverless={mode}` or run `yarn serverless-{mode}`
 valid modes are currently: `es`, `oblt`, and `security`
 
 configuration is applied in the following order, later values override
- 1. kibana.yml
- 2. serverless.yml
- 3. serverless.{mode}.yml
+ 1. serverless.yml (serverless configs go first)
+ 2. serverless.{mode}.yml (serverless configs go first)
+ 3. base config, in this preference order:
+  - my-config.yml(s) (set by --config)
+  - env-config.yml (described by `env.KBN_CONFIG_PATHS`)
+  - kibana.yml (default @ `env.KBN_PATH_CONF`/kibana.yml)
```

Comment:
cc @afharo given he knows the problem more than I do: Are we fine with changing this ordering?

Reply:
Just to add, this change reflects the status quo as it is on `main` now for (1, 2), and (3) is the one whose order is currently in regression, trying to be fixed.

`main` now: 
```
/Users/alex/Git/kibana/config/serverless.yml
/Users/alex/Git/kibana/config/serverless.es.yml
/Users/alex/Git/kibana/config/kibana.yml
/Users/alex/Git/kibana/config/my-config.yml
/Users/alex/Git/kibana/config/serverless.recent.yml
/Users/alex/Git/kibana/config/kibana.dev.yml
/Users/alex/Git/kibana/config/serverless.recent.dev.yml
```
on this branch:
```
/Users/alex/Git/kibana/config/serverless.yml
/Users/alex/Git/kibana/config/serverless.es.yml
/Users/alex/Git/kibana/config/my-config.yml
/Users/alex/Git/kibana/config/serverless.recent.yml
/Users/alex/Git/kibana/config/kibana.dev.yml
/Users/alex/Git/kibana/config/serverless.recent.dev.yml
```